asapdiscovery-docking/asapdiscovery/docking/docking_v2.py

```diff
@@ -0,0 +1,283 @@
+    score_type: SCORE_TYPE = Field(description="Docking score type")
+    score: float = Field(description="Docking score")
+    provenance: dict[str, str] = Field(description="Provenance")
+
+    @root_validator
+    @classmethod
+    def smiles_match(cls, values):
+        posed_ligand = values.get("posed_ligand")
+        input_pair = values.get("input_pair")
+        if posed_ligand.smiles != input_pair.ligand.smiles:
+            raise ValueError(
+                "SMILES of ligand and ligand in input docking pair not match"
+            )
+        return values
+
+    def to_posed_oemol(self) -> oechem.OEMol:
+        """
+        Combine the target and ligand into a single oemol
+        """
+        _, prot, _ = split_openeye_design_unit(self.input_pair.complex.target.to_oedu())
+        return combine_protein_ligand(prot, self.posed_ligand.to_oemol())
+
+
+class DockingBase(BaseModel):
+    """
+    Base class for docking.
+    """
+
+    type: Literal["DockingBase"] = "DockingBase"
+
+    @abc.abstractmethod
+    def _dock() -> list[DockingResult]:
+        ...
+
+    def dock(
+        self, inputs: list[DockingInputPair], use_dask: bool = False, dask_client=None
+    ) -> Union[list[dask.delayed], list[DockingResult]]:
+        if use_dask:
+            delayed_outputs = []
+            for inp in inputs:
+                out = dask.delayed(self._dock)(inputs=[inp])
+                delayed_outputs.append(out[0])  # flatten
+            outputs = actualise_dask_delayed_iterable(
+                delayed_outputs, dask_client=dask_client
+            )
+        else:
+            outputs = self._dock(inputs=inputs)
+        return outputs
+
+    @abc.abstractmethod
+    def provenance(self) -> dict[str, str]:
+        ...
+
+
+class POSIT_METHOD(Enum):
+    ALL = oedocking.OEPositMethod_ALL
+    HYBRID = oedocking.OEPositMethod_HYBRID
+    FRED = oedocking.OEPositMethod_FRED
+    MCS = oedocking.OEPositMethod_MCS
+    SHAPEFIT = oedocking.OEPositMethod_SHAPEFIT
+
+    @classmethod
+    def reverse_lookup(cls, value):
+        return cls(value).name
+
+
+class POSIT_RELAX_MODE(Enum):
+    CLASH = oedocking.OEPoseRelaxMode_CLASHED
+    ALL = oedocking.OEPoseRelaxMode_ALL
+    NONE = oedocking.OEPoseRelaxMode_NONE
+
+
+class POSITDocker(DockingBase):
+    """
+    Docker class for POSIT.
+    """
+
+    type: Literal["POSITDocker"] = "POSITDocker"
+    score_type: Literal[SCORE_TYPE.CHEMGAUSS4] = SCORE_TYPE.CHEMGAUSS4
+
+    relax: POSIT_RELAX_MODE = Field(
+        POSIT_RELAX_MODE.NONE,
+        description="When to check for relaxation either, 'clash', 'all', 'none'",
+    )
+    posit_method: POSIT_METHOD = Field(
+        POSIT_METHOD.ALL, description="POSIT method to use"
+    )
+    use_omega: bool = Field(True, description="Use omega to generate conformers")
+    num_poses: PositiveInt = Field(1, description="Number of poses to generate")
+    log_name: str = Field("run_docking_oe", description="Name of the log file")
```

Comment:
We could probably drop this and just absorb the log file into the result object in a field like stdout?

Reply:
Yep sounds good